plugins/repository-gcs/src/main/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageBlobStore.java

```diff
@@ -197,55 +197,22 @@ boolean blobExists(String blobName) throws IOException {
      * @return an InputStream
      */
     InputStream readBlob(String blobName) throws IOException {
-        return SocketAccess.doPrivilegedIOException(() -> {
-            
+        for(int retry = 0; retry < 3; retry++){
             try {
-                logger.debug("Reading blob called " + blobName);
-                Storage.Objects.Get object = client.objects().get(bucket, blobName);
-                return object.executeMediaAsInputStream();
-            } catch (GoogleJsonResponseException e) {
-                GoogleJsonError error = e.getDetails();
-                if ((e.getStatusCode() == HTTP_NOT_FOUND) || ((error != null) && (error.getCode() == HTTP_NOT_FOUND))) {
-                    throw new NoSuchFileException(e.getMessage());
-                }
-                logger.warn("[repository-gcs][readBlob]Quota exceeded - Throttling 1st try");
+                return SocketAccess.doPrivilegedIOException(() -> {
+                    logger.debug("Reading blob called " + blobName);
+                    Storage.Objects.Get object = client.objects().get(bucket, blobName);
+                    return object.executeMediaAsInputStream();
+                });
+            }catch(Exception e){
                 
-                TimeUnit.SECONDS.sleep(2);
-                return readBlob(blobName,1);
+                logger.warn("[ReadBlob()] Lost - Throttling try #" + retry );
```

Comment:
Same thing here - we should throw something on the last retry so that the user knows that something actually failed

Reply:
will add a commit today